src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,45 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+
+
+public class Duke {
+    private FileUpdater fileUpdater;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Constructor for Duke class
+     * @param filePath of a file which consists of a list of tasks to be loaded by Duke
+     */
+    public Duke(String filePath){
+        tasks = new TaskList();
+        ui = new Ui(tasks);
+        try {
+            File f = new File(filePath);
+            this.fileUpdater = new FileUpdater(f, tasks);
```

Comment:
Perhaps it is better to use a word for variable names (like ```file``` instead of ```f```). I noticed the same issue in several other places too (and also for parameters for some of your methods eg. ```String s``` can be something like ```String input```).